src/main/java/org/jenkinsci/plugins/compressbuildlog/BuildLogCompressor.java

```diff
@@ -55,57 +58,29 @@ public boolean isApplicable(Class<? extends Job> jobType) {
 
         @Override
         public void onFinalized(Run run) {
-            File log = run.getLogFile();
-            if (log.getName().endsWith(".gz")) {
-                // ignore already compressed log
-                LOGGER.log(Level.FINER, String.format("Skipping %s because the log is already compressed", run));
-                return;
-            }
 
             if (run.getParent().getProperty(BuildLogCompressor.class) == null) {
                 LOGGER.log(Level.FINER, String.format("Skipping %s because the project is not configured to have compressed logs", run));
                 return;
             }
 
-            String gzippedLogName = log.getName() + ".gz";
-
-            if (log.getName().equals("log")) {
-                LOGGER.log(Level.FINE, String.format("Compressing build log of %s", run));
-
-                // regular, expected log file
-                FileInputStream fis = null;
-                FileOutputStream fos = null;
-                GZIPOutputStream gzos = null;
+            Util.compressLogFile(run);
 
+            Plugin mvn = Jenkins.getInstance().getPlugin("maven-plugin");
```

Comment:
Couldn't the plugin be disabled, or newly installed and enabled only after a restart?